tools/importer-rest-api-specs/components/dataapigeneratoryaml/templater_constants.go

```diff
@@ -0,0 +1,77 @@
+package dataapigeneratoryaml
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/go-yaml/yaml"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/pandora/tools/sdk/resourcemanager"
+)
+
+type Constant struct {
+	Name   string  `yaml:"Name"`
+	Type   string  `yaml:"Type"`
+	Values []Value `yaml:"Values"`
+}
+
+type Value struct {
+	Name        string `yaml:"Name"`
+	Description string `yaml:"Description"`
+}
+
+func codeForConstant(constantName string, details resourcemanager.ConstantDetails) ([]byte, error) {
+	sortedKeys := make([]string, 0)
+	for key := range details.Values {
+		sortedKeys = append(sortedKeys, key)
+	}
+	sort.Strings(sortedKeys)
+
+	var values []Value
```

Comment:
in the case there's no constants, `values` would be nil here which has a different value to empty when marshalled:

```suggestion
	values := make([]Value, 0)
```